lib/Runtime/Library/BuiltIn/IndexOf.js

```diff
@@ -38,7 +38,7 @@
     }
 
     while (k < len) {
-        if (o[k] === searchElement && (o[k] !== undefined || searchElement !== undefined)) {
+        if (o[k] === searchElement && (o[k] !== undefined || (searchElement === undefined && o.hasOwnProperty(k)))) {
```

Comment:
It's worth writing a test case like @MikeHolman said. I think `k in o` might be the only correct way to get `undefined` at indexes in the prototype which would otherwise look like holes.